trigger/project_triggers.go

```diff
@@ -103,33 +103,32 @@ func metadataFromVersion(args ProcessorArgs) (model.VersionMetadata, error) {
 		Alias:               args.Alias,
 	}
 	if args.SourceVersion != nil {
+		repo, err := model.FindRepository(args.DownstreamProject.Id)
+		if err != nil {
+			return metadata, errors.Wrap(err, "finding most recent revision")
+		}
+		if repo == nil {
+			return metadata, errors.Errorf("repo '%s' not found", args.DownstreamProject.Id)
```

Comment:
If we require a source version, then it seems correct that we'd require the repository, right? Otherwise we hit the else case of the push trigger? I might be misunderstanding how that trigger works though, please DM me if that isn't the case. 

I am inclined to allow this to be configured even when there are no versions, since that's a state we should allow for the push version kind of trigger, and users may want to set things up before enabling the project (i.e. no versions will exist at that time).

Reply:
Discussed offline - the `LastRevision` field is actually needed for all types of triggers, and the source version is the a version on the upstream project, whereas the repo revision query is done on the downstream project, so they are exclusive (i.e. source version can exist even if there are no downstream versions).

Reply:
Tweaked; I'm a little on the fence about whether we should have this validation at all or just keep it an error since it's so niche and isn't really possible to validate for repos, but seems worth telling users when we can.